### ISSUE TITLE
Truffle decoder bug

### ISSUE THREAD

**Issue by fredlacs** [trufflesuite/truffle#5053]
## Issue

Truffle decoder hits an unexpected `undefined` when parsing a tx on rinkeby.

## Steps to Reproduce

When attempting to bridge eth from ethreum to arbitrum, the truffle decoder isn't able to parse the [`outboundTransfer` function
](https://github.com/OffchainLabs/arbitrum/blob/b749cfa9597d149fa22b41000b093387687937e0/packages/arb-bridge-peripherals/contracts/tokenbridge/libraries/gateway/ITokenGateway.sol#L42-L49)

You can reproduce this by going to https://bridge.arbitrum.io/ and trying to bridge eth while connected to the rinkeby network

Here is the tx where this happened https://rinkeby.etherscan.io/tx/0x5309d0f7fa7b2b1f8c85617fbbbd22217d88eae2065ef555d46a3003cf6c8117

## Expected Behavior

Expected the truffle decoder to appropriately handle the function provided

## Actual Results

"Cannot read property of undefined"
![telegram-cloud-photo-size-4-5958365184819050304-y](https://user-images.githubusercontent.com/32464905/165104649-3c9f8171-1fed-47d9-9cb9-d90a4717355f.jpg)

**Comment by haltman-at** [comment]
So it looks like you're reporting an issue with MetaMask's Transaction Insight, rather than Truffle Decoder directly?  That said, it sure *looks* like an issue with Truffle Decoder, I have to admit.  On the other hand, I can't seem to reproduce this using only `@truffle/decoder` and `@truffle/fetch-and-compile`.  So, I don't know what's wrong here.  Have you tried [reporting this to MetaMask](https://github.com/MetaMask/metamask-extension/issues/new?assignees=&labels=bug&template=bug-report.yml&title=%5BBug%5D%3A+)?  Do you have any further reproduction information?

I can try reproducing this some more later, but for now I'm not encountering an issue when I just use Truffle Decoder directly, rather than as part of Transaction Inisght.

**Comment by haltman-at** [comment]
Hm, hold on, the output that I'm getting doesn't seem right, even if it's not the same problem you're getting.  I guess I'll go back and take a second look at this.  That said, you still might want to try reporting this to MetaMask.

**Comment by haltman-at** [comment]
Yeah, after fixing the problem I was getting due to setting things up wrong, so I'm actually getting a sensible decoding now, I'm still not getting the problem using Truffle Decoder alone.  This may be a problem with MetaMask; not sure.

**Comment by sukanyaparashar** [comment]
Hi @fredlacs . Can you please let us know if you are still facing this issue?
